sequence_processing_pipeline/Pipeline.py

```diff
@@ -365,15 +398,35 @@ def _validate_mapping_file(mapping_file_path):
                'target_subfragment', 'primer', 'primer_plate', 'sample_name',
                'run_center', 'primer_date', 'target_gene', 'processing_robot',
                'extractionkit_lot'}
+
         try:
             df = pd.read_csv(mapping_file_path, delimiter='\t')
 
-            # if the two sets of headers are equal, then return the dataframe
-            # and no error message.
-            if len(exp - set(df.columns)) == 0:
-                return df
+            # rename all columns to their lower-case versions.
+            # we will want to return this version to the user.
+            df.columns = df.columns.str.lower()
 
-            msg = 'missing columns: %s' % ', '.join(exp - set(df.columns))
+            # if the two sets of headers are equal, verify there are no
+            # duplicate sample-names and then return the dataframe and no
+            # error message.
+            if len(exp - set(df.columns)) == 0:
+                sample_names = df['sample_name'].tolist()
+                dupes = Counter(sample_names)
+                dupes = [x for x in dupes if dupes[x] > 1]
```

Comment:
Just realized that you could do this directly with pandas: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.value_counts.html

Reply:
nice! ty for the reference; that's much better!